### DISCUSSION TITLE
Drag resize window issue

### DISCUSSION THREAD

**Discussion by xland** [mikke89/RmlUi#327] [Q&A] [answered]
This is what happens when I change a window from small to large by drag the windiow's border:
![image](https://user-images.githubusercontent.com/2980915/174932580-055d100a-cf08-4f6b-b92d-adca484de239.png)
When I release the mouse, The border shadows disappear.
Does this problem have anything to do with my custom window borders?
```c++
case WM_NCCALCSIZE: {
    return 0;
}
case WM_NCHITTEST:
{
    return windowHitTest(window_handle, l_param);
    break;
}
```
Windows 10 System

**Comment by mikke89** [comment, marked as answer]
I suspect this is essentially the same reason as in #325: You need to re-draw manually during the resize operation with the mouse. This is more of a platform API question, Win32 in your case. I don't know the specific solution here but there should be plenty of resources on this topic when searching for it.

I'm sure it could be useful to others so if you do find a specific solution then please share it here :)

**Reply by xland** [reply]
I already included `Context::SetDimensions()` and `ElementDocument::UpdateDocument()` in my code when this problem occurred.

```c++
	case WM_SIZE:
	{
		// Intercept sizing to set the OpenGL viewport, then submit it to the platform handler for context sizing.
		const int width = LOWORD(l_param);
		const int height = HIWORD(l_param);
		data->window_dimensions.x = width;
		data->window_dimensions.y = height;
		data->render_interface.SetViewport(width, height);
		if (data->context)
			data->context->SetDimensions(data->window_dimensions);
		return 0;
	}
```

```c++
	case Rml::EventId::Resize: {
		document->UpdateDocument();		
		setHitTestArea();
		break;
	}
```
I've tried to find a solution, but I can't find one,
I will keep trying. Once I find a solution, I will share it here.

**Reply by mikke89** [reply]
Since event processing stalls during the resize, you also need to render the context during the `WM_SIZE` call and swap the window buffer. This works for me in the win32 backend:

```cpp
	case WM_SIZE:
	{
		const int width = LOWORD(l_param);
		const int height = HIWORD(l_param);
		data->window_dimensions.x = width;
		data->window_dimensions.y = height;
		data->render_interface.SetViewport(width, height);
		if (data->context)
		{
			data->context->SetDimensions(data->window_dimensions);
			data->context->Update();

			Backend::BeginFrame();
			data->context->Render();
			Backend::PresentFrame();
		}
		return 0;
	}
	break;
```

See also [this answer](https://stackoverflow.com/a/56614042) for a related GLFW approach.

**Reply by xland** [reply]
Yes，It works.
Thanks very much.
